cmd/release-controller/upgrades.go

```diff
@@ -317,34 +322,173 @@ func syncGraphToSecret(graph *UpgradeGraph, update bool, secretClient kv1core.Se
 		}
 		return true, nil
 	}, stopCh)
+}
 
-	if !update {
+func saveUpgradeGraph(buf *bytes.Buffer, graph *UpgradeGraph, secretClient kv1core.SecretInterface, ns, name string) {
+	if err := graph.Save(buf); err != nil {
+		klog.Errorf("Unable to calculate graph state: %v", err)
+		return
+	}
+	secret, err := secretClient.Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Can't read latest secret %s/%s: %v", ns, name, err)
 		return
 	}
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte)
+	}
+	secret.Data["latest"] = buf.Bytes()
+	if _, err := secretClient.Update(context.TODO(), secret, metav1.UpdateOptions{}); err != nil {
+		klog.Errorf("Can't save state to secret %s/%s: %v", ns, name, err)
```

Comment:
You should be returning errors in this method and then logging the error outside

Reply:
Done